src/routes/profile/ProfileView.vue

```diff
@@ -0,0 +1,161 @@
+
+        <v-virtual-scroll
+          v-if="games.length > 0"
+          :items="games"
+          height="600"
+          item-height="100"
+          @scroll="handleScroll"
+        >
+          <template #default="{ item }">
+            <ProfileGameListItem
+              :name="item.name"
+              :is-ranked="item.isRanked"
+              :winner-label="getIsWinner(item)"
+              :opponent-name="getOpponentName(item)"
+              @replay="goToReplay(item.id)"
+            />
+          </template>
+        </v-virtual-scroll>
+
+        <!-- Loading indicator -->
+        <div v-if="myGamesStore.loading" class="text-center pa-4">
+          <v-progress-circular indeterminate color="primary" />
+        </div>
+
+        <!-- Fallback message when no games -->
+        <p v-else-if="games.length === 0" style="color: rgba(var(--v-theme-surface-1))">
+          {{ t('profile.noGamesFound') }}
+        </p>
+      </v-card>
+    </v-container>
+  </div>
+</template>
+
+<script setup>
+import { computed, onMounted, onUnmounted } from 'vue';
+import { useAuthStore } from '@/stores/auth';
+import DiscordLink from '@/components/DiscordLink.vue';
+import { useMyGamesStore } from '@/stores/myGames';
+import ProfileGameListItem from './ProfileGameListItem.vue';
+import { useI18n } from 'vue-i18n';
+
+const { t } = useI18n();
+const authStore = useAuthStore();
+const myGamesStore = useMyGamesStore();
+
+const hasDiscord = computed(() =>
+  authStore.identities?.some(({ provider }) => provider === 'discord')
+);
+const discordUsername = computed(() => {
+  const discordIdentity = authStore.identities?.find(({ provider }) => provider === 'discord');
+  return discordIdentity?.username || '';
+});
+
+const games = computed(() => myGamesStore.games);
+
+function getOpponentName(game) {
+  const opponent = game.p0.isOpponent ? game.p0 : game.p1;
+  return opponent.username;
+}
+
+function getIsWinner(game) {
+  if (!game.winnerId) {
+    return null;
+  }
+  const opponent = game.p0.isOpponent ? game.p0 : game.p1;
+  return game.winnerId !== opponent.id;
+}
+
+async function fetchGames() {
+  try {
+    await myGamesStore.loadMyGames({
+      sortBy: 'createdAt',
+      sortDirection: 'desc',
+      reset: true
+    });
+  } catch (error) {
+    console.error('Failed to fetch games:', error);
+  }
+}
+
+function handleScroll(event) {
+  const { scrollTop, scrollHeight, clientHeight } = event.target;
+  const scrollThreshold = 200;
+
+  if (scrollHeight - scrollTop - clientHeight < scrollThreshold) {
+    loadMoreGames();
+  }
+}
+
+async function loadMoreGames() {
+  if (myGamesStore.loading || !myGamesStore.hasMore) {return;}
+
+  try {
+    await myGamesStore.loadMyGames({
+      sortBy: 'createdAt',
+      sortDirection: 'desc'
+    });
+  } catch (error) {
+    console.error('Failed to load more games:', error);
+  }
+}
+
+function goToReplay(gameId) {
+  window.location.href = `/spectate/${gameId}`;
```

Comment:
Let's use the Vue router for navigation. We can import useRouter and instantiate it at the top then this becomes a router.push() call